### PR TITLE
Analytics: Make visible trigger accept an array of visible percentages

### PR DESCRIPTION
**Pull request by zhouyx**
For #11986 
I change the design according to https://github.com/ampproject/amphtml/issues/11873#issuecomment-342974876

The config should looks like:
```
visiblePercentageThresholds: [[10, 100]]
```

We will only handle `visiblePercentageThresholds` if neither `visiblePercentageMin` nor `visiblePercentageMax` are present. Please let me know : )

### CODE REVIEW

File: extensions/amp-analytics/0.1/visibility-manager.js [hunk 1]
```diff
@@ -230,11 +231,69 @@ export class VisibilityManager {
...
+   * @param {!Element=} opt_element
+   * @return {!UnlistenDef}
+   */
+  createModelAndListen_(calcVisibility, spec,
+      readyPromise, createReportPromiseFunc, callback, opt_element) {
+    if (spec['visiblePercentageThresholds'] &&
+        spec['visiblePercentageMin'] == undefined &&
+        spec['visiblePercentageMax'] == undefined) {
+      const unlisteners = [];
+      //const ranges = tryParseJson(spec['visiblePercentageThresholds']);
```
Comment (jonkeller) [inline extensions/amp-analytics/0.1/visibility-manager.js]: Nit: Remove commented-out code

File: extensions/amp-analytics/0.1/visibility-manager.js [hunk 2]
```diff
@@ -230,11 +231,69 @@ export class VisibilityManager {
...
+      const ranges = spec['visiblePercentageThresholds'];
+      if (!ranges || !isArray(ranges)) {
+        user().error(TAG, 'invalid visiblePercentageThresholds');
+      }
+      console.log('ranges is ', ranges);
```
Comment (jonkeller) [inline extensions/amp-analytics/0.1/visibility-manager.js]: Nit: Remove console.log() calls? Also, may want to return after user().error() above.

File: extensions/amp-analytics/0.1/visibility-manager.js [hunk 3]
```diff
@@ -230,11 +231,69 @@ export class VisibilityManager {
...
+        }
+        if (!isFiniteNumber(percents[0]) || !isFiniteNumber(percents[1])) {
+          // not valid number
+          user().error(TAG,
+              'visiblePercentageThresholds entry is not valid number');
+          continue;
+        }
+        const min = Number(percents[0]);
+        const max = Number(percents[1]);
+        if (min < 0 || max > 100 || min >= max) {
```
Comment (jonkeller) [inline extensions/amp-analytics/0.1/visibility-manager.js]: Why can't min == max?

File: extensions/amp-analytics/0.1/visibility-manager.js [hunk 4]
```diff
@@ -20,10 +20,14 @@ import {
   nativeIntersectionObserverSupported,
 } from '../../../src/intersection-observer-polyfill';
 import {VisibilityModel} from './visibility-model';
-import {dev} from '../../../src/log';
+import {dev, user} from '../../../src/log';
 import {getMode} from '../../../src/mode';
 import {map} from '../../../src/utils/object';
 import {Services} from '../../../src/services';
+import {tryParseJson} from '../../../src/json';
```
[the remark this answers is not in the document]
Comment (zhouyx) [reply extensions/amp-analytics/0.1/visibility-manager.js]: removed

File: extensions/amp-analytics/0.1/visibility-manager.js (hunk 1 above)
Comment (zhouyx) [reply extensions/amp-analytics/0.1/visibility-manager.js]: done

File: extensions/amp-analytics/0.1/visibility-manager.js (hunk 2 above)
Comment (zhouyx) [reply extensions/amp-analytics/0.1/visibility-manager.js]: good point. done

File: extensions/amp-analytics/0.1/visibility-manager.js (hunk 3 above)
Comment (zhouyx) [reply extensions/amp-analytics/0.1/visibility-manager.js]: IntersectionObserver require the visiblePercentage to be greater than min, and less or equal to max. With `min==max`, we would not be able to fire any visible event.

Comment (jonkeller) [reply extensions/amp-analytics/0.1/visibility-manager.js]: Ah, I see. Thank you for clarifying!

**Pull request merged**